### PR TITLE
OS-26 Location list page modifications

### PR DESCRIPTION
**Pull request by rotator**
**Steps for review**
After installation or update OpenY profile you can check modified settings form in OpenY Map module /admin/config/services/openy_map with all settings for Location list page and explanations how to add new location content type to Locations search.

@podarok Please, review

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Comment by podarok** [comment]
ok to test

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy.cibox.tools/build308 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build308 |
| Installation process | http://install.openy-dev.ffwua.com/build308/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/308/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build308/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build308/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build308/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build308/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build308/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy.cibox.tools/build311 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build311 |
| Installation process | http://install.openy-dev.ffwua.com/build311/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/311/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build311/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build311/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build311/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build311/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build311/blogWCAG2AAhtmlcs.txt

File: modules/custom/location_finder/src/Plugin/Block/LocationFinder.php [hunk 1]
```diff
@@ -13,29 +17,88 @@
...
-        '#name' => 'locations',
-        '#display_id' => 'locations_camps_block',
-      ],
-      [
-        '#type' => 'view',
-        '#name' => 'locations',
-        '#display_id' => 'locations_facilities_block',
-      ],
-    ];
+    // Location view
```
Comment (podarok) [inline modules/custom/location_finder/src/Plugin/Block/LocationFinder.php]: Could you setup PHPStorm to be using Drupal code style formatting?

Comment (podarok) [reply modules/custom/location_finder/src/Plugin/Block/LocationFinder.php]: Should be fullstop

File: modules/custom/location_finder/src/Plugin/Block/LocationFinder.php [hunk 2]
```diff
@@ -13,29 +17,88 @@
...
+    $locationsView = 'locations';
+    $locationDisplay = 'locations_block';
+
+    // Render Locations block display with changed arguments
```
Comment (podarok) [inline modules/custom/location_finder/src/Plugin/Block/LocationFinder.php]: Fullstop, please

File: modules/custom/openy_map/src/Form/SettingsForm.php [hunk 3]
```diff
@@ -4,12 +4,48 @@
...
+  /**
+   * Class constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Config factory.
+   * @param \Drupal\openy_map\OpenyMapManager $openy_map_manager
+   *   The OpenY Map manager.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory,
+                              OpenyMapManager $openy_map_manager) {
```
Comment (podarok) [inline modules/custom/openy_map/src/Form/SettingsForm.php]: This looks like not a Drupal code style. Could you fix it?

File: modules/custom/openy_map/src/OpenyMapManager.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  /**
+   * Class constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   The entity field manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager,
+                              EntityFieldManagerInterface $entity_field_manager) {
```
Comment (podarok) [inline modules/custom/openy_map/src/OpenyMapManager.php:36]: Not a Drupal code style, please fix

File: modules/custom/openy_map/src/OpenyMapManager.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+   */
+  public function getLocationNodeTypes() {
+    $fieldMap = $this->entityFieldManager->getFieldMap();
+    $locationBundles = $fieldMap['node']['field_location_coordinates']['bundles'];
+    $nodeTypes = $this->entityTypeManager->getStorage('node_type')->loadByProperties(['type' => array_keys($locationBundles)]);
+
+    return $nodeTypes;
+  }
+
+}
```
Comment (podarok) [inline modules/custom/openy_map/src/OpenyMapManager.php]: New line please, needed everywhere for better git diff
https://www.drupal.org/docs/develop/standards/coding-standards#indenting

**Review by podarok — CHANGES_REQUESTED** [review]
Code needs some love

**Review by podarok — DISMISSED** [review]

**Comment by podarok** [comment]
@kolesnikoff is it possible to get video how to use this feature?
It looks like highly complicated from the step by step description

**Comment by gundevel** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy.cibox.tools/build330 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build330 |
| Installation process | http://install.openy-dev.ffwua.com/build330/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/330/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build330/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build330/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build330/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build330/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build330/blogWCAG2AAhtmlcs.txt

**Comment by podarok** [comment]
image
Icons should respect installation in subfolder.
Please fix

File: modules/custom/openy_map/src/Form/SettingsForm.php [hunk 6]
```diff
@@ -31,22 +66,92 @@ protected function getEditableConfigNames() {
...
+          $path = '/' . $themePath . '/' . $themeFile;
+          $fileOptions[$path] = '<img src="' . $path . '" />';
+        }
+      }
+      $locationFinderPath = drupal_get_path('module', 'location_finder') . '/img';
+      $locationFinderFiles = array_values(array_diff(scandir($locationFinderPath), ['.', '..']));
+      foreach ($locationFinderFiles as $locFile) {
+        if (!in_array($locFile, $themeFiles)) {
+          $path = '/' . $locationFinderPath . '/' . $locFile;
+          $fileOptions[$path] = '<img src="' . $path . '" />';
```
Comment (podarok) [inline modules/custom/openy_map/src/Form/SettingsForm.php]: Images should respect subfolder installation

**Review by podarok — CHANGES_REQUESTED** [review]

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy.cibox.tools/build342 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build342 |
| Installation process | http://install.openy-dev.ffwua.com/build342/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/342/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build342/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build342/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build342/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build342/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build342/blogWCAG2AAhtmlcs.txt

**Comment by podarok** [comment]
Functionality looks like not working
image
image

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy.cibox.tools/build343 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build343 |
| Installation process | http://install.openy-dev.ffwua.com/build343/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/343/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build343/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build343/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build343/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build343/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build343/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy.cibox.tools/build346 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build346 |
| Installation process | http://install.openy-dev.ffwua.com/build346/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/346/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build346/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build346/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build346/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build346/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build346/blogWCAG2AAhtmlcs.txt

**Comment by podarok** [comment]
Go go go @kolesnikoff

**Pull request merged**